api/lib/validate.js

```diff
@@ -166,8 +178,95 @@ function validateOrderbook(orderbook) {
   }
 }
 
+function validateSettings(settings) {
+  if (typeof settings !== 'object') {
+    throw error('Invalid parameter: settings');
```

Comment:
Promises handle exceptions much nicer than callback functions, but, then `api` should have a promise-returning interface and forego callbacks if the callbacks won't always get passed the error